File: backend/src/main/java/com/example/intern_dashboard/service/InternService.java
package com.example.intern_dashboard.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.example.intern_dashboard.dto.request.InternRequest;
import com.example.intern_dashboard.dto.response.InternResponse;
import com.example.intern_dashboard.enums.InternStatus;

public interface InternService {

    Page<InternResponse> getAllInterns(
            int page,
            int size,
            String sortBy,
            String direction
    );

    InternResponse getInternById(Integer id);

    InternResponse createIntern(InternRequest dto);

    InternResponse updateIntern(
            Integer id,
            InternRequest dto
    );

    void deleteIntern(Integer id);

    List<InternResponse> searchInterns(String keyword);

    Page<InternResponse> filterInterns(
            InternStatus status,
            String department,
            int page,
            int size,
            String sortBy,
            String direction
    );
}

File: frontend/src/services/internService.js
import axios from "axios";

const API_URL =
    "http://localhost:8080/api/interns";

export async function getAllInterns(
    page = 0,
    size = 5,
    sortBy = "id",
    direction = "asc"
) {

    const response =
        await axios.get(
            `${API_URL}?page=${page}&size=${size}&sortBy=${sortBy}&direction=${direction}`
        );

    return response.data;
}

export async function searchInterns(
    keyword,
    page = 0,
    size = 5,
    sortBy = "id",
    direction = "asc"
) {

    const response =
        await axios.get(
            `${API_URL}/search?keyword=${keyword}&page=${page}&size=${size}&sortBy=${sortBy}&direction=${direction}`
        );

    return response.data;
}

export async function createIntern(
    internData
) {

    const response =
        await axios.post(
            API_URL,
            internData
        );

    return response.data;
}

export async function updateIntern(
    id,
    internData
) {

    const response =
        await axios.put(
            `${API_URL}/${id}`,
            internData
        );

    return response.data;
}

export async function deleteIntern(id) {

    const response =
        await axios.delete(
            `${API_URL}/${id}`
        );

    return response.data;
}